master/internal/trials/authz_iface.go

```diff
@@ -0,0 +1,53 @@
+package trials
+
+import (
+	"context"
+
+	"github.com/uptrace/bun"
+
+	"github.com/determined-ai/determined/master/internal/authz"
+	"github.com/determined-ai/determined/master/pkg/model"
+	"github.com/determined-ai/determined/proto/pkg/projectv1"
+)
+
+// TrialsAuthZ describes authz methods for experiments.
+type TrialAuthZ interface {
+
+	// POST /trial-comparison/collections
+	CanCreateTrialCollection(
+		ctx context.Context, curUser *model.User, p *projectv1.Project,
```

Comment:
Unless you expect, down the road, to use more data from the Project in your decision other than its id, I recommend changing this API to accept a project id rather than an entire Project struct. I think this line:
```go
AuthZProvider.Get().CanCreateTrialCollection(ctx, curUser, &req.ProjectId)
```
looks a lot better than:
```go
AuthZProvider.Get().CanCreateTrialCollection(ctx, curUser, &projectv1.Project{Id: req.ProjectId})
```
This would also be a clearer signal to users of your API what data is actually used to decide stuff; they wouldn't need to wonder which fields they need to actually fill in on the Project struct.